src/components/LinkThresholdFacilitiesToGroupModal.vue

```diff
@@ -11,45 +11,95 @@
   </ion-header>
 
   <ion-content>
-    <ion-searchbar />
-
-    <ion-radio-group>
-      <ion-item>
-        <ion-radio>
-          <ion-label>
-            {{ "Facility name" }}
-            <p>{{ "Facility ID" }}</p>
-          </ion-label>
-        </ion-radio>
-      </ion-item>
-      <ion-item>
-        <ion-radio>
-          <ion-label>
-            {{ "Facility name" }}
-            <p>{{ "Facility ID" }}</p>
-          </ion-label>
-        </ion-radio>
-      </ion-item>
-      <ion-item>
-        <ion-radio>
+    <ion-radio-group v-if="configFacilities.length" v-model="selectedFacilityId">
+      <ion-item v-for="facility in configFacilities" :key="facility.facilityId">
+        <ion-radio :value="facility.facilityId">
           <ion-label>
-            {{ "Facility name" }}
-            <p>{{ "Facility ID" }}</p>
+            {{ facility.facilityName }}
+            <p>{{ facility.facilityId }}</p>
           </ion-label>
         </ion-radio>
       </ion-item>
     </ion-radio-group>
 
+    <div v-else class="empty-state">
+      <p>{{ translate("No facility found.") }}</p>
+    </div> 
+
     <ion-fab vertical="bottom" horizontal="end" slot="fixed">
-      <ion-fab-button>
+      <ion-fab-button @click="saveFacility()">
         <ion-icon :icon="saveOutline" />
       </ion-fab-button>
     </ion-fab>
   </ion-content>
 </template>
 
 <script setup lang="ts">
-import { IonButton, IonButtons, IonContent, IonFab, IonFabButton, IonHeader, IonIcon, IonItem, IonLabel, IonRadio, IonRadioGroup, IonSearchbar, IonTitle, IonToolbar } from "@ionic/vue";
+import { IonButton, IonButtons, IonContent, IonFab, IonFabButton, IonHeader, IonIcon, IonItem, IonLabel, IonRadio, IonRadioGroup, IonTitle, IonToolbar, modalController } from "@ionic/vue";
 import { closeOutline, saveOutline } from "ionicons/icons";
 import { translate } from '@/i18n'
-</script>
\ No newline at end of file
+import { useStore } from "vuex";
+import { computed, defineProps, onMounted, ref } from "vue";
+import { hasError, showToast } from "@/utils";
+import { ChannelService } from '@/services/ChannelService';
+import logger from "@/logger";
+import { DateTime } from "luxon";
+
+const store = useStore();
+const selectedFacilityId = ref("");
+const props = defineProps(["group", "selectedConfigFacilityId"]);
+
+const configFacilities = computed(() => store.getters["util/getConfigFacilities"])
+
+onMounted(() => {
+  selectedFacilityId.value = props.selectedConfigFacilityId?.facilityId ? JSON.parse(JSON.stringify(props.selectedConfigFacilityId.facilityId)) : '';
+})
+
+function closeModal() {
+  modalController.dismiss({ dismissed: true });
+}
+
+async function saveFacility() {
+  if(!selectedFacilityId.value) {
```

Comment:
Add check to only update when there is some change in facilityId

Reply:
Improved code to disable save button until config facility updated.